pallets/msa/src/types.rs

```diff
@@ -13,6 +13,12 @@ use scale_info::TypeInfo;
 /// Dispatch Empty
 pub const EMPTY_FUNCTION: fn(MessageSourceId) -> DispatchResult = |_| Ok(());
 
+///For table testing patterns
+pub struct TestCase<T> {
```

Comment:
NB: the types file is just for production code - you could put this struct in `tests.rs`, or else you could put it in `mocks.rs`

Thanks!